src/lexer.d

```diff
@@ -42,7 +44,25 @@ class Token
 
     pure Token clone()
     {
-        return new Token(this.line, this.type, this.value);
+        return new Token(this.line, this.lineOffset, this.type, this.value);
+    }
+}
+
+struct ProgrammerExceptionInfo
```

Comment:
Why are these params in a separate struct from ProgrammerException?

Reply:
Eh no good reason at this point.

Reply:
I don't really mind it, just curious

Reply:
Dropped it